src/neo4j_graphrag/experimental/components/schema.py

```diff
@@ -236,3 +333,81 @@ async def run(
             SchemaConfig: A configured schema object, constructed asynchronously.
         """
         return self.create_schema_model(entities, relations, potential_schema)
+
+
+class SchemaFromTextExtractor(Component):
+    """
+    A component for constructing SchemaConfig objects from the output of an LLM after
+    automatic schema extraction from text.
+    """
+
+    def __init__(
+        self,
+        llm: LLMInterface,
+        prompt_template: Optional[PromptTemplate] = None,
+        llm_params: Optional[Dict[str, Any]] = None,
+    ) -> None:
+        self._llm: LLMInterface = llm
+        self._prompt_template: PromptTemplate = (
+            prompt_template or SchemaExtractionTemplate()
+        )
+        self._llm_params: dict[str, Any] = llm_params or {}
+
+    @validate_call
+    async def run(self, text: str, examples: str = "", **kwargs: Any) -> SchemaConfig:
+        """
+        Asynchronously extracts the schema from text and returns a SchemaConfig object.
+
+        Args:
+            text (str): the text from which the schema will be inferred.
+            examples (str): examples to guide schema extraction.
+        Returns:
+            SchemaConfig: A configured schema object, extracted automatically and
+            constructed asynchronously.
+        """
+        prompt: str = self._prompt_template.format(text=text, examples=examples)
+
+        response = await self._llm.ainvoke(prompt, **self._llm_params)
+        content: str = (
+            response
+            if isinstance(response, str)
```

Comment:
The `LLMInterface` `invoke` and `ainvoke` methods always return an `LLMResponse` object, so you can assume you can access `response.content` directly (if no `LLMGenerationError` is raised)